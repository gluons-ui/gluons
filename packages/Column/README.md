## Mandala UI Column Component
Mandala is a React component library that utilizes a declarative CSS library for styling.

### Installation

Check the [MonoRepo README](https://github.com/mandala-ui/mandala) for installation.
### Description
A Column is used for separating content from other content. Defaults to 100% width for you to layout how you prefer.
### Usage
if using the whole library:
```js
import { Column } from 'mandala';
```
if only using this component:
```js
import Column from '@mandala-ui/column';
```

### Props
* `backgroundColor` - background color string
* `borderColor` - border color string
* `borderRadius` - number clamped between 1 and 4
* `children` - Content inside the column
* `padding` - number clamped between 1 and 4, padding around all

| propName | propType | defaultValue | isRequired |
| -------- | -------- | ------------ | ---------- |
| backgroundColor| string     | white| -          |
| borderColor    | string     | white| -          |
| borderRadius   | number     | 0    | -          |
| children       | node       | null | -          |
| padding        | number     | 0    | -          |
| radius         | number     | 0    | -          |

#### TODO:
- [ ] Border Width

![mandala-bottom](https://user-images.githubusercontent.com/1824267/38281372-32114e5c-375f-11e8-9fbe-e5191b96429c.png)
