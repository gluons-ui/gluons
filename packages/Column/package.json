{
  "author": "Lucas K Allmon",
  "description": "Mandala UI Column Component",
  "keywords": [
    "react",
    "css",
    "library",
    "components",
    "interface",
    "functional",
    "OOCSS",
    "declarative",
    "ui",
    "user",
    "ux"
  ],
  "license": "MIT",
  "main": "lib/Column.js",
  "module": "es/Column.js",
  "name": "@mandala-ui/column",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "prepare": "nwb build-react-component"
  },
  "version": "1.0.2"
}
