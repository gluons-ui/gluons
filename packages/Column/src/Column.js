import React from 'react';
import PropTypes from 'prop-types';

const Column = ({
  borderColor,
  backgroundColor,
  children,
  padding,
  radius }) => {
  const borders = `${borderColor ? `ba b--${borderColor}` : ''}`;
  return (
    <div className="column w-100">
      <div className={`${borders} bg-${backgroundColor} br${radius} pa${padding} w-100`}>
        {children || null}
      </div>
    </div>
  );
};

Column.propTypes = {
  borderColor: PropTypes.string,
  backgroundColor: PropTypes.string,
  children: PropTypes.node,
  padding: PropTypes.oneOf([0, 1, 2, 3, 4, 5]),
  radius: PropTypes.oneOf([0, 1, 2, 3, 4]),
};

Column.defaultProps = {
  borderColor: null,
  backgroundColor: 'white',
  children: null,
  padding: 2,
  radius: 0,
};

export default Column;
