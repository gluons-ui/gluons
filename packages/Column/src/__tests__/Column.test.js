import React from 'react';
import { shallow } from 'enzyme';
import Column from '../Column';

const testContent = (
  <div>
    <span>lorem ipsum</span>
    <span>lorem ipsum</span>
    <span>lorem ipsum</span>
    <span>lorem ipsum</span>
    <span>lorem ipsum</span>
    <span>lorem ipsum</span>
  </div>
);

describe('<Column />', () => {
  const component = shallow(
    <Column>
      {testContent}
    </Column>,
  );

  it('exists', () => {
    expect(component).toBeDefined();
  });

  it('renders defaults correctly', () => {
    expect(shallow(
      <Column />,
    )).toMatchSnapshot();
  });

  it('can render without children', () => {
    expect(component.find('button').children()).toHaveLength(0);
  });

  it('renders a custom background color', () => {
    expect(shallow(
      <Column
        backgroundColor="gray"
      >
        {testContent}
      </Column>,
    )).toMatchSnapshot();
  });

  it('renders a custom border radius', () => {
    expect(shallow(
      <Column
        radius={3}
      >
        {testContent}
      </Column>,
    )).toMatchSnapshot();
  });

  it('renders a custom border color', () => {
    expect(shallow(
      <Column
        borderColor="red"
      >
        {testContent}
      </Column>,
    )).toMatchSnapshot();
  });

  it('renders a custom padding', () => {
    expect(shallow(
      <Column
        borderColor="red"
      >
        {testContent}
      </Column>,
    )).toMatchSnapshot();
  });
});
