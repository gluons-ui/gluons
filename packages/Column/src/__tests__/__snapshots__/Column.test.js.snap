// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Column /> renders a custom background color 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`<Column /> renders a custom border color 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`<Column /> renders a custom border radius 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`<Column /> renders a custom padding 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`<Column /> renders defaults correctly 1`] = `
ShallowWrapper {
  "length": 1,
}
`;
